### ISSUE TITLE
Filter events by date in CMS

### ISSUE THREAD

**Issue by saumier** [culturecreates/footlight-app#103]
![Image](https://user-images.githubusercontent.com/419491/214296377-3c0060c5-fdc2-402f-9f25-853e0a030012.png)

**Comment by troughc** [comment]
I just looked through the basic options and if, for a date range you click hte same date twice, the range is that one day. On the other hand, the old cms has a date picker already, that allows you to click one date, click another, drag to select a rabge of dates in calendar. Is that not an option? @saumier

**Comment by saumier** [comment]
You answered my question.  We will use the AntD widget for date range. Thx.

**Comment by troughc** [comment]
@saumier FYI, Maude again mentioned the need to filter for past events:
"I need a way to differentiate between past and upcoming events. For me it's essential, sometimes I have events that repeat themselves (like story time event in 2022... or when I'm looking for old info, etc.)

**Comment by AbhishekPAnil** [comment]
ESTIMATE  **3.5** hrs

Subtasks : 
1.Add calendar component to filter.
2.Add the date filter to event filter api.

**Comment by AbhishekPAnil** [comment]
ESTIMATE : **2** hrs 

Subtasks : 
1.Add event status and corresponding color tags to event list page.
2.Add virtual location .

**Comment by troughc** [comment]
@AbhishekPAnil 
Here is the blue indicator for when the date filter has been applied. Stephanie added it to the figma. 

![Image](https://user-images.githubusercontent.com/64489312/222749040-40feb218-bd75-4641-9bac-9848e68639bd.png)

In the same section as this


![Image](https://user-images.githubusercontent.com/64489312/222749136-0bb4e976-e4de-4c9a-bcad-053d8a1e1a46.png)

**Comment by AbhishekPAnil** [comment]
@SuhailAliyar Wil need support in backend to get the details of all the events created in a calendar and also for past events from the mentioned date.

**Comment by SuhailAliyar** [comment]
@saumier We need to modify the event filter (only for CMS API). I propose updating the filter as follows:
All: startDate and EndDate : Null
Past: startDate: Null and EndDate: Yesterday's date.

**Comment by saumier** [comment]
Proposal based on discussion this morning. Note that in the URL the query parameter can have "param=" or just "param" without the equal when there is no value. I have given both examples below.

### upcoming events (default)
?page=1  
-->` {"page": "1"}`
Note: when start-date-span in not defined, assume today.

### all events 
 ?page=1&start-date-span&end-date-span  
 ?page=1&start-date-span=&end-date-span=  
--> `{"page": "1", "start-date-span": null, "end-date-span": null}`

### events in date span
?page=1&start-date-span=2023-01-01&end-date-span=2023-04-30  
--> `{"page": "1", "start-date-span": "2023-01-01", "end-date-span": "2023-04-30"}`

### upcoming events before a date (today is 2023-03-09)
?page=1&end-date-span=2023-03-08
--> `{"page": "1",  "end-date-span":"2023-03-08"}`
Note: when start-date-span in not defined, assume today.

### all events before a date (today is 2023-03-09)
?page=1&start-date-span&end-date-span=2023-03-08 
?page=1&start-date-span=&end-date-span=2023-03-08
--> `{"page": "1", "start-date-span": null, "end-date-span":"2023-03-08"}`

### all events after a date
?page=1&start-date-span=2023-01-01
 --> `{"page": "1", "start-date-span": "2023-01-01"}`

?page=1&start-date-span=2023-01-01&end-date-span  
?page=1&start-date-span=2023-01-01&end-date-span=
 --> `{"page": "1", "start-date-span": "2023-01-01", "end-date-span":null}`

**Comment by saumier** [comment]
@SuhailAliyar Please take a look at the proposal above.  I found this relevant article about sending param=%00 as an explicit null value that is percent encoded. But I didn't keep that idea in the proposal above.  

How will we use Swagger UI to test this, since it doesn't let us send a blank param?

**Comment by saumier** [comment]
@troughc Please test in staging.

**Comment by troughc** [comment]
@saumier, @AbhishekPAnil  
Tested Staging in chrome, logged in as a guest user. 
I found a few issues:
- I set the range from March 15-30 and got an event for March 31.   See screenshot:
<img width="885" alt="Screenshot 2023-03-15 at 1 10 51 PM" src="https://user-images.githubusercontent.com/64489312/225388284-5347701c-1166-454c-bbbf-1f44341b1caf.png">
Or I select a single date, the example I'm using for a single date is March 30. This March 31st event is included in the result: http://staging.cms.footlight.io/dashboard/62df90a4820e41f7c1359760/events/640eca7fe5cde30064142a65

- If I do all time, i get a blank screen. See screenshot:
<img width="678" alt="Screenshot 2023-03-15 at 1 12 10 PM" src="https://user-images.githubusercontent.com/64489312/225388561-5d146e7a-7eb4-4d79-8241-ec17db5fdaa5.png">-

- As a workaround: I was able to set a time in the past and in the future to get events from all time. 
- The past events option worked for me.
